cloud/scope/machine_pool.go

```diff
@@ -571,17 +628,313 @@ func (m *MachinePoolScope) TerminateInstancePool(ctx context.Context, instancePo
 
 // instancePoolNeedsUpdates compares incoming OCIMachinePool and compares against existing instance pool.
 func instancePoolNeedsUpdates(machinePoolScope *MachinePoolScope, instancePool *core.InstancePool) bool {
+	instanePoolSize := 0
+	if instancePool.Size != nil {
+		instanePoolSize = *instancePool.Size
+	}
+	if int(*machinePoolScope.MachinePool.Spec.Replicas) != instanePoolSize {
+		return true
+	} else if !(reflect.DeepEqual(machinePoolScope.OCIMachinePool.Spec.InstanceConfiguration.InstanceConfigurationId, instancePool.InstanceConfigurationId)) {
+		return true
+	}
+	return false
+}
 
-	// Allow pool resize
-	if machinePoolScope.MachinePool.Spec.Replicas != nil {
-		if instancePool.Size == nil || int(*machinePoolScope.MachinePool.Spec.Replicas) != *instancePool.Size {
-			return true
+func (m *MachinePoolScope) getAgentConfig() *core.InstanceConfigurationLaunchInstanceAgentConfigDetails {
+	agentConfigSpec := m.OCIMachinePool.Spec.InstanceConfiguration.AgentConfig
+	if agentConfigSpec != nil {
+		agentConfig := &core.InstanceConfigurationLaunchInstanceAgentConfigDetails{
+			IsMonitoringDisabled:  agentConfigSpec.IsMonitoringDisabled,
+			IsManagementDisabled:  agentConfigSpec.IsManagementDisabled,
+			AreAllPluginsDisabled: agentConfigSpec.AreAllPluginsDisabled,
 		}
-	} else if instancePool.Size != nil {
-		return true
+		if len(agentConfigSpec.PluginsConfig) > 0 {
+			pluginConfigList := make([]core.InstanceAgentPluginConfigDetails, len(agentConfigSpec.PluginsConfig))
+			for i, pluginConfigSpec := range agentConfigSpec.PluginsConfig {
+				pluginConfigRequest := core.InstanceAgentPluginConfigDetails{
+					Name: pluginConfigSpec.Name,
+				}
+				desiredState, _ := core.GetMappingInstanceAgentPluginConfigDetailsDesiredStateEnum(string(pluginConfigSpec.DesiredState))
```

Comment:
If the value isn't found should the `bool` be checked error? or set a default or something?

Reply:
fixed